app/Plugin/LinOTPAuth/Controller/Component/Auth/LinOTPAuthenticate.php

```diff
@@ -21,6 +28,76 @@ public function authenticate(CakeRequest $request, CakeResponse $response)
         return $user;
     }
 
+    /*
+     * Query LinOTP
+     */
+    private static function _linotp_verify($baseUrl, $realm, $user, $password, $verifyssl)
+    {
+        $params = array();
+        $params['ssl_allow_self_signed'] = !$verifyssl;
+        $params['ssl_verify_peer_name'] = $verifyssl;
+        $params['ssl_verify_peer'] = $verifyssl;
+
+        $HttpSocket = new HttpSocket($params);
+
+        // POST data
+        $data = array(
+            "user" => $user,
+            "pass" => $password,
+            "realm" => $realm,
+        );
+
+        $url = "$baseUrl/validate/check";
+
+        CakeLog::debug( "Sending POST request to ${url}");
+        $results = $HttpSocket->post($url, $data);
+        $response = json_decode($results->body());
```

Comment:
Before this line I'd check if the response was succesful, you can do this via $results->code() or $results->isOk() iirc. This will avoid accidentally parsing / trying to json_decode() unwanted responses.

Reply:
Thanks for the note - back to xdebug to check if all is fine...